MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -640,75 +649,84 @@ - (void)createAnalyzerFile {
 - (void)setupLogBuffer {
   @synchronized(self) {
 
-    // Array of 20 buffer file paths.
-    NSArray<NSString *> *bufferFiles = [self createLogBufferFilesIfNeeded];
+    // Array of 20 buffer file paths per priority.
+    // Each priority has 0.mscrasheslogbuffer ... 19.mscrasheslogbuffer.
+    for (NSInteger priority = 0; priority < kMSPriorityCount; priority++) {
+      NSArray *files = [self createBufferFilesIfNeededForPriority:(MSPriority)priority];
 
-    // Just to make sure we have max of 20 files on disk. If we have less than 20 files on disk, we can't really do
-    // anything. This case should never happen.
-    NSInteger count = bufferFiles.count >= ms_crashes_log_buffer_size ? ms_crashes_log_buffer_size : bufferFiles.count;
+      // Init the counter for each priority.
+      [self.bufferIndex setObject:@0 forKey:@(priority)];
 
-    // Initialize the bufferIndex and write pre-fill the filepaths of the MSCrashesBufferedLog
-    self.bufferIndex = 0;
-    for (NSUInteger i = 0; i < count; i++) {
-      MSCrashesBufferedLog emptyLog = MSCrashesBufferedLog{bufferFiles[i], nil};
-      msCrashesLogBuffer[i] = emptyLog;
+      // Create a buffer for the priority. Making use of `{}` as we're using C++11.
+      msCrashesLogBuffer[(MSPriority)priority][{}];
```

Comment:
While it works, using the "`[]` implicitly emplaces" form is technically executing a void expression for its side effects. Prefer an explicit insert or emplace - `msCrashesLogBuffer.insert(static_cast<MSPriority>(priority), {});`. Also, due to that same side effect, this initialization is unnecessary anyhow; it will happen automatically the first time through the loop below.